packages/microservices/client/client-kafka.ts

```diff
@@ -94,46 +94,56 @@ export class ClientKafka extends ClientProxy {
     this.consumer && (await this.consumer.disconnect());
     this.producer = null;
     this.consumer = null;
+    this.initialized$ = null;
     this.client = null;
   }
 
   public async connect(): Promise<Producer> {
     if (this.client) {
-      return this.producer;
+      return this.initialized$.then(() => this.producer);
```

Comment:
Another question around consistency: 
in the rest of the function, we use async await but in there we leverage `.then`.

I'd suggest keeping the same approach here to make the code more readable even if the execution is not a 100% replacement of the promise chaining.
```suggestion
      await this.initialized$;
      return this.producer;
```

Reply:
`.then` is intentionally used here to indicate the intent. The Promise here is used for slightly different purpose than other promises in this class